File: backend/app/schemas/wedding_trial.py
from pydantic import BaseModel
from typing import List, Optional


class TrialStartRequest(BaseModel):
    paket: str = "basic"


class TrialStartResponse(BaseModel):
    session_id: str
    paket: str
    expires_at: str


class Step1Request(BaseModel):
    session_id: str
    kota: str
    paket: str = "basic"


class Step1Response(BaseModel):
    session_id: str
    kota: str
    paket: str
    kota_tersedia: bool


class Step2BudgetRequest(BaseModel):
    session_id: str
    budget_total: int


class PriceItemSchema(BaseModel):
    item_name: str
    kategori: str
    prioritas: str
    harga_estimasi_min: int
    harga_estimasi_max: int
    harga_alokasi: int
    bisa_diskip: bool

    class Config:
        from_attributes = True


class Step2BudgetResponse(BaseModel):
    session_id: str
    budget_total: int
    rata_rata_kota_min: int
    rata_rata_kota_max: int
    status_budget: str
    items_wajib: List[PriceItemSchema]
    items_penting: List[PriceItemSchema]
    items_opsional: List[PriceItemSchema]


class Step2KonsepRequest(BaseModel):
    session_id: str
    konsep: str


class BudgetTierSchema(BaseModel):
    label: str
    budget_total: int
    status_budget: str
    items_wajib: List[PriceItemSchema]
    items_penting: List[PriceItemSchema]
    items_opsional: List[PriceItemSchema]


class Step2KonsepResponse(BaseModel):
    session_id: str
    konsep: str
    nama_referensi: str
    deskripsi_singkat: Optional[str]
    tiers: List[BudgetTierSchema]


class TrialSessionResponse(BaseModel):
    session_id: str
    paket: str
    kota: str
    jalur: Optional[str]
    budget_total: Optional[int]
    konsep_pilihan: Optional[str]
    created_at: str
    expires_at: str

    class Config:
        from_attributes = True

File: frontend/src/types/weddingTrial.ts
export type Paket = "basic" | "premium" | "vendor_tracker";

export interface TrialStartResponse {
  session_id: string;
  paket: Paket;
  expires_at: string;
}

export interface Step1Response {
  session_id: string;
  kota: string;
  paket: Paket;
  kota_tersedia: boolean;
}

export interface PriceItem {
  item_name: string;
  kategori: string;
  prioritas: "wajib" | "penting" | "opsional";
  harga_estimasi_min: number;
  harga_estimasi_max: number;
  harga_alokasi: number;
  bisa_diskip: boolean;
}

export interface Step2BudgetResponse {
  session_id: string;
  budget_total: number;
  rata_rata_kota_min: number;
  rata_rata_kota_max: number;
  status_budget: "terbatas" | "normal" | "leluasa";
  items_wajib: PriceItem[];
  items_penting: PriceItem[];
  items_opsional: PriceItem[];
}

export interface BudgetTier {
  label: "Basic" | "Ideal" | "Premium";
  budget_total: number;
  status_budget: "terbatas" | "normal" | "leluasa";
  items_wajib: PriceItem[];
  items_penting: PriceItem[];
  items_opsional: PriceItem[];
}

export interface Step2KonsepResponse {
  session_id: string;
  konsep: string;
  nama_referensi: string;
  deskripsi_singkat: string | null;
  tiers: BudgetTier[];
}

export interface TrialSession {
  session_id: string;
  paket: Paket;
  kota: string;
  jalur: "budget" | "konsep";
  budget_total: number | null;
  konsep_pilihan: string | null;
  created_at: string;
  expires_at: string;
}
